esx_service/utils/vmdk_utils.py

```diff
@@ -109,21 +122,21 @@ def get_datastores():
     'url' is datastore URL (e.g. '/vmfs/volumes/vsan:572904f8c031435f-3513e0db551fcc82')
     'dockvol-path; is a full path to 'dockvols' folder on datastore
     """
-    if datastores == None:
-        init_datastoreCache()
+    init_datastoreCache()
     return datastores
```

Comment:
No, thread A has finished init_datastoreCache() and is going to return "datastores" and thread B calls init_datastoreCache() and sets datastores to None. Thread A will be returning None.